### PR TITLE
Bluetooth: Controller: Add support for PAwR advertiser

### PR DESCRIPTION
**Pull request by hermabe**
Includes changes needed to support Periodic Advertising with Responses - Advertiser in addition to other fixes and improvements to the softdevice controller.

### CODE REVIEW

File: subsys/bluetooth/controller/hci_driver.c [hunk 1]
```diff
@@ -644,6 +655,13 @@ static int configure_supported_features(void)
 		}
 	}
 
+	if (IS_ENABLED(CONFIG_BT_CTLR_SDC_PAWR_ADV)) {
```
Comment (olivier-le-sage) [inline subsys/bluetooth/controller/hci_driver.c:658]: I'd prefer if we used #if defined(CONFIG_BT_CTLR_SDC_PAWR_ADV) here too

File: subsys/bluetooth/controller/Kconfig [hunk 2]
```diff
@@ -310,5 +311,53 @@ config BT_UNINIT_MPSL_ON_DISABLE
...
+	int "Periodic Advertising with Responses - Advertiser sets"
+	range 0 10
+	default 1
+	help
+	  Maximum number of concurrent Periodic Advertising with Responses
+	  advertisers supported.
+
+config BT_CTLR_SDC_PERIODIC_ADV_RSP_TX_BUFFER_COUNT
+	int "Number of PAwR advertiser TX buffers"
+	range 0 15
```
Comment (olivier-le-sage) [inline subsys/bluetooth/controller/Kconfig:334]:
```suggestion
	range 1 15
```

0 isn't allowed

File: subsys/bluetooth/controller/Kconfig [hunk 3]
```diff
@@ -310,5 +311,53 @@ config BT_UNINIT_MPSL_ON_DISABLE
...
+	  Maximum size of subevent indications. Decreasing this will reduce memory
+	  usage and increase scheduling performance. The controller will disallow
+	  sending a subevent indication with a larger payload.
+
+config BT_CTLR_SDC_PERIODIC_ADV_RSP_RX_BUFFER_COUNT
+	int "Number of PAwR advertiser RX buffers"
+	range 0 255
+	default 2
+	help
+	  Maximum number of subevent responses that can be buffered at a time.
```
Comment (olivier-le-sage) [inline subsys/bluetooth/controller/Kconfig:355]:
```suggestion
	  Maximum number of periodic advertising response reports that can be buffered at a time.
```

(I imagine we'll store reports for failed RX in the ring buffer frames too)

File: subsys/bluetooth/controller/hci_driver.c (hunk 1 above)
Comment (hermabe) [reply subsys/bluetooth/controller/hci_driver.c:658]: `IS_ENABLED` is used for all the others. The call to `sdc_support_le_periodic_adv_with_rsp` is optimized away and the linker will not include anything PAwR-related unless enabled.

**Review by olivier-le-sage — COMMENTED** [review]
Not a Kconfig expert. Do we need to add a BT_CTLR_SDC_PAWR_ADV_SUPPORT (or similar) ? Looking at line 11

File: subsys/bluetooth/controller/Kconfig [hunk 4]
```diff
@@ -310,5 +311,53 @@ config BT_UNINIT_MPSL_ON_DISABLE
 	  mpsl_lib_uninit, which releases all peripherals and allows the user to
 	  override any interrupt handlers used by MPSL.
 
+config BT_CTLR_SDC_PAWR_ADV
+	bool "LE Periodic Advertising with Responses Advertising State [EXPERIMENTAL]"
+	select EXPERIMENTAL
+	select BT_CTLR_CHAN_SEL_2
```
Comment (olivier-le-sage) [inline subsys/bluetooth/controller/Kconfig:317]: I think we get BT_CTLR_CHAN_SEL_2 through BT_CTLR_ADV_PERIODIC

**Comment by hermabe** [comment]
> Not a Kconfig expert. Do we need to add a BT_CTLR_SDC_PAWR_ADV_SUPPORT (or similar) ? Looking at line 11

Ideally these kconfigs would be in zephyr, but there is no support for PAwR there yet. When host support is added we would move these configs there. Then we would have the same pattern here as for the other features:
- One config applications use for enabling host support: `BT_FEATURE`
- One config for enabling controller support: `BT_CTLR_FEATURE`
- One config that controllers select to indicate support: `BT_CTLR_FEATURE_SUPPORT`.

The `BT_CTLR_FEATURE_SUPPORT` is there so that if you select a controller that does not support the feature you will get an error/warning if you enable `BT_FEATURE`. For now there is only one controller in play so I think we can wait with all of this until it is moved to zephyr.

File: subsys/bluetooth/controller/Kconfig (hunk 2 above)
Comment (laxiLang) [reply subsys/bluetooth/controller/Kconfig]: why 15? any theory

Comment (olivier-le-sage) [reply subsys/bluetooth/controller/Kconfig]: What if the host uses multiple commands? The count parameter of data requests is limited to the number of subevents, so max 128 -- I'd imagine someone could want to allocate more tx buffers to reduce the amount of traffic over HCI

Comment (hermabe) [reply subsys/bluetooth/controller/Kconfig]: Can move the upper limit to 128. I guess the device will run out of memory before it hits those limits anyway.

**Comment by hermabe** [comment]
Rebased on main and included the needed changes from nrfxlib

**Review by rugeGerritsen — APPROVED** [review]

**Review by olivier-le-sage — APPROVED** [review]

**Review by wiba-nordic — APPROVED** [review]

**Review by KyraLengfeld — APPROVED** [review]

**Pull request merged**